hc-dpki/src/keypair.rs

```diff
@@ -0,0 +1,528 @@
+use crate::{
+    bundle,
+    holochain_sodium::{aead, kx, random::random_secbuf, secbuf::SecBuf, sign},
+    util,
+};
+use holochain_core_types::{agent::KeyBuffer, error::HolochainError};
+use rustc_serialize::json;
+use std::str;
+
+pub struct Keypair {
+    pub pub_keys: String,
+    pub sign_priv: SecBuf,
+    pub enc_priv: SecBuf,
+}
+
+pub const SEEDSIZE: usize = 32 as usize;
+
+impl Keypair {
+    /// derive the pairs from a 32 byte seed buffer
+    ///  
+    /// @param {SecBuf} seed - the seed buffer
+    pub fn new_from_seed(seed: &mut SecBuf) -> Self {
+        let mut seed = seed;
+        let mut sign_public_key = SecBuf::with_insecure(sign::PUBLICKEYBYTES);
+        let mut sign_secret_key = SecBuf::with_secure(sign::SECRETKEYBYTES);
+        let mut enc_public_key = SecBuf::with_insecure(kx::PUBLICKEYBYTES);
+        let mut enc_secret_key = SecBuf::with_secure(kx::SECRETKEYBYTES);
+
+        sign::seed_keypair(&mut sign_public_key, &mut sign_secret_key, &mut seed).unwrap();
+        kx::seed_keypair(&mut seed, &mut enc_public_key, &mut enc_secret_key).unwrap();
```

Comment:
think i saw a comment to this effect already, but just in case. Prefer returning Results throughout and using `?` instead of unwrap.

Reply:
I think so, we want our error handling to be as friction-less as possible